src/main/kotlin/no/nav/syfo/minesykmeldte/MineSykmeldteMapper.kt

```diff
@@ -0,0 +1,59 @@
+package no.nav.syfo.minesykmeldte
+
+import no.nav.syfo.kafka.felles.SykepengesoknadDTO
+import no.nav.syfo.minesykmeldte.db.SykmeldtDbModel
+import no.nav.syfo.minesykmeldte.model.PreviewSoknad
+import no.nav.syfo.minesykmeldte.model.PreviewSykmelding
+import no.nav.syfo.model.sykmelding.arbeidsgiver.ArbeidsgiverSykmelding
+import no.nav.syfo.model.sykmelding.arbeidsgiver.SykmeldingsperiodeAGDTO
+import no.nav.syfo.model.sykmelding.model.GradertDTO
+import no.nav.syfo.model.sykmelding.model.PeriodetypeDTO
+import java.time.LocalDate
+
+class MineSykmeldteMapper private constructor() {
+    companion object {
+        fun toPreviewSykmelding(sykmeldingDbModel: SykmeldtDbModel): PreviewSykmelding {
+            return PreviewSykmelding(
+                id = sykmeldingDbModel.sykmeldingId,
+                fom = sykmeldingDbModel.sykmelding.sykmeldingsperioder.minOf { it.fom },
+                tom = sykmeldingDbModel.sykmelding.sykmeldingsperioder.maxOf { it.tom },
+                type = getTypeSykmelding(sykmeldingDbModel.sykmelding),
+                lest = sykmeldingDbModel.lestSykmelding
+            )
+        }
+
+        fun toPreviewSoknad(soknad: SykepengesoknadDTO, lest: Boolean): PreviewSoknad {
+            return PreviewSoknad(
+                id = soknad.id,
+                sykmeldingId = soknad.sykmeldingId,
+                fom = soknad.fom,
+                tom = soknad.tom,
+                status = soknad.status.name,
+                sendtDato = soknad.sendtArbeidsgiver?.toLocalDate(),
+                lest = lest
+            )
+        }
+
+        private fun getTypeSykmelding(sykmelding: ArbeidsgiverSykmelding): String {
+            val now = LocalDate.now()
+            val sortedPeriods = sykmelding.sykmeldingsperioder.sortedBy { it.fom }
+
+            return when (val periodNearestNow = sortedPeriods.find { it.tom >= now }) {
```

Comment:
Det her var veldig smart! 🏆 Hadde ikke engang tenkt på denne problemstillingen, kjempebra at dere tok høyde for at sykmeldingen kan ha flere perioder med ulik grad :)

Reply:
Hmm, det kan kanskje hende Bendik vil vise alle periodene hvis sykmeldingen har mer enn en periode? Kanskje lurt å høre med han?